### PR TITLE
Redux store

### PR DESCRIPTION
**Pull request by lorenzoworx**
In this step, I implement Redux Toolkit.
### Changes Made
- Add Redux Toolkit using (`npm install react-redux @reduxjs/toolkit`).
- Create a directory that contains all Redux logic (`/src/redux`)
- Configure a Redux store (`/src/redux/store.js`)
- Define a `slice` of state for `books` that:
   - Includes an **array of books** (initial state: empty array)
   - Includes a **reducer that adds a book**
   - Includes a **reducer that removes a book**
- Define a `slice` of state for `categories` that:
   - Includes an **array of categories** (initial state: empty array)
   - Includes a **reducer that checks the status** and always returns "Under construction" (the initial state should check to that string)
- Implement a 'feature folder' file structure

### CODE REVIEW

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { configureStore } from '@reduxjs/toolkit';
+
+const store = configureStore({
+  reducer: {},
+});
```
Comment (binhussen) [inline src/redux/store.js]: - Kindly, Use configurestore to create reducers for `books`, and `categories` 👍 
look the bellow code snippet 👇
```jsx
const store = configureStore({
  reducer: { books: booksReducer, categories: categoriesReducer },
})
```

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  books: [],
+};
+
+export const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books.concat(action.payload);
+    },
```
Comment (binhussen) [inline src/redux/books/booksSlice.js]: - Kindly `concat()` method does not mutate the original array but instead returns a new array populated with elements of all the arrays that were merged, so here you are not assigning the returned value to the book state, please assign it to not lose what you added 👍 

`state.books = state.books.concat(action.payload);`

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @lorenzoworx , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Used react-redux and toolkit✔️
- Well Structured Code ✔️
- Used Git flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @lorenzoworx,

## APPROVED 🟢

## Highlights
✔️ Slices for reducers

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**